lib/Conversion/ImportVerilog/Expressions.cpp

```diff
@@ -22,6 +22,37 @@ struct ExprVisitor {
   ExprVisitor(Context &context, Location loc)
       : context(context), loc(loc), builder(context.builder) {}
 
+  /// Helper function to convert a value to its simple bit vector
+  /// representation, if it has one. Otherwise returns null.
+  Value convertToSimpleBitVector(Value value) {
+    if (!value)
+      return {};
+    if (auto type = dyn_cast_or_null<moore::UnpackedType>(value.getType())) {
+      if (type.isSimpleBitVector())
+        return value;
+      if (auto sbvt = type.castToSimpleBitVectorOrNull())
+        return builder.create<moore::ConversionOp>(
+            loc, sbvt.getType(builder.getContext()), value);
+    }
+    mlir::emitError(loc, "expression of type ")
+        << value.getType() << " cannot be cast to a simple bit vector";
+    return {};
+  }
+
+  /// Helper function to convert a value to its "truthy" boolean value.
+  Value convertToBool(Value value) {
+    if (!value)
+      return {};
+    if (auto type = dyn_cast_or_null<moore::IntType>(value.getType()))
+      if (type.getBitSize() == 1)
+        return value;
+    if (auto type = dyn_cast_or_null<moore::UnpackedType>(value.getType()))
+      return builder.create<moore::BoolCastOp>(loc, value);
+    mlir::emitError(loc, "expression of type ")
+        << value.getType() << " cannot be cast to a boolean";
```

Comment:
I'm not entirely sure how to trigger these two. Slang's type checker should prevent this from ever appearing in the AST. But I'm also not sure if `ImportVerilog` covers all cases, and it seemed better to have a defensive error here since this is user-facing.